devicetypes/terence/zxt-600-zwave-to-ir-extender.src/zxt-600-zwave-to-ir-extender.groovy

```diff
@@ -0,0 +1,1167 @@
+        if (degreesInteger < 17) {
+            degreesInteger = 17
+        }
+        // ZXT-600 highest setting is 28 C
+        if (degreesInteger > 30) {
+            degreesInteger = 30
+        }
+    } else {
+        // ZXT-600 lowest settings is 67 F
+        if (degreesInteger < 63) {
+            degreesInteger = 63
+        }
+        // ZXT-600 highest setting is 28
+        if (degreesInteger > 86) {
+            degreesInteger = 86
+        }
+    }
+    log.debug "setCoolingSetpoint({$degreesInteger} ${temperatureScale})"
+    sendEvent("name":"coolingSetpoint", "value":degreesInteger, unit: "C")
+    setThermostatMode("cool")
+}
+
+
+//***** Set the thermostat */
+def setThermostatSetpoint(degrees) {
+    log.debug "setThermostatSetpoint called.....want to get rid of that"
+    // convert the temperature to a number and execute
+    setThermostatSetpoint(degrees.toDouble())
+}
+
+// Configure
+// Syncronize the device capabilities with those that the UI provides
+def configure() {
+ 	// update the device's remote code to ensure it provides proper mode info
+    // The reason why remote set code process is excluded from the below sequence is 
+    // the time consuming is up to 12s based on ZXT-600 characteristic.
+
+    setRemoteCode()
+}
+
+/*
+    Strongly recommend to reserve 30s delay for all parameter SET of configuration command class.
+*/
+private secureConfigSequence(commands, delay=3000) {
+	delayBetween(commands.collect{ secure(it) }, delay)
+}
+
+// Switch Fan Mode
+// Switch to the next available fan speed
+def switchFanMode() {
+    // Determine the current fan speed setting
+    def currentMode = device.currentState("thermostatFanMode")?.value
+    def lastTriedMode = getDataByName("lastTriedFanMode") ?: currentMode.value ?: "off"
+
+    // Determine what fan speeds are available
+    def supportedModes = getDataByName("supportedFanModes") ?: "fanAuto fanLow"
+    def modeOrder = fanModes()
+    //log.info modeOrder
+
+    // Determine what the next fan speed should be
+    def next = { modeOrder[modeOrder.indexOf(it) + 1] ?: modeOrder[0] }
+    def nextMode = next(lastTriedMode)
+    while (!supportedModes?.contains(nextMode) && nextMode != "fanAuto") {
+        nextMode = next(nextMode)
+    }
+
+    // Make it so
+    switchToFanMode(nextMode)
+}
+
+// Switch to Fan Mode
+// Given the name of a fan mode, make it happen
+def switchToFanMode(nextMode) {
+    def supportedFanModes = getDataByName("supportedFanModes")
+    if(supportedFanModes && !supportedFanModes.tokenize()?.contains(nextMode)) log.warn "thermostat mode '$nextMode' is not supported"
+
+    // If the mode is even possible
+    if (nextMode in fanModes()) {
+        // Try to switch to the mode
+        updateState("lastTriedFanMode", nextMode)
+        return "$nextMode"()  // Call the function perform the mode switch
+    } else {
+        // Otherwise, bail
+        log.debug("no fan mode method '$nextMode'")
+    }
+}
```

Comment:
Instead of that, you can just utilize [airConditionerFanMode](https://developer-preview.smartthings.com/docs/devices/capabilities/capabilities-reference/#airConditionerFanMode) capability and its command.